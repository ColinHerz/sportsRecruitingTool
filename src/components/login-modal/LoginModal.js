import React, { useState } from "react";

import "./login-modal.scss";

const LoginModal = props => {
	const [isRegistering, setIsRegistering] = useState(props.isRegistering);
	const [email, setEmail] = useState(``);
	const [password, setPassword] = useState(``);

	const registering = event => {
		event.preventDefault();
		setIsRegistering(true);
	};

	const loggingIn = event => {
		event.preventDefault();
		setIsRegistering(false);
	};

	const handleEmailChange = event => {
		event.preventDefault();

		setEmail(event.target.value);
	};

	const handlePasswordChange = event => {
		event.preventDefault();

		setPassword(event.target.value);
	};

	const handleSubmit = event => {
		event.preventDefault();
	};

	return (
		<div id="login-modal">
			<h2>Sporta</h2>

			<button onClick={loggingIn}>Log In</button>

			<button onClick={registering}>Register</button>

			<form onSubmit={handleSubmit}>
				<label>
					Email
					<input
						type="text"
						value={email}
						onChange={handleEmailChange}
					/>
				</label>

				<label>
					Password
					<input
						type="password"
						value={password}
						onChange={handlePasswordChange}
					/>
				</label>

				<input type="submit" value="Submit" />
			</form>
		</div>
	);
};

export default LoginModal;
