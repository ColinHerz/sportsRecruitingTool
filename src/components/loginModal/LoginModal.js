import PropTypes from "prop-types";
import React, { useState } from "react";

import "./loginModal.scss";

const LoginModal = props => {
	const [isRegistering, setIsRegistering] = useState(props.isRegistering);
	const [email, setEmail] = useState(``);
	const [password, setPassword] = useState(``);

	const registering = event => {
		event.preventDefault();
		setIsRegistering(true);
	};

	const loggingIn = event => {
		event.preventDefault();
		setIsRegistering(false);
	};

	const handleEmailChange = event => {
		event.preventDefault();

		setEmail(event.target.value);
	};

	const handlePasswordChange = event => {
		event.preventDefault();

		setPassword(event.target.value);
	};

	const handleSubmit = event => {
		event.preventDefault();
	};

	return (
		<div id="gray-background">
			<div id="login-modal">
				<h2>Sporta</h2>

				<div id="btn-sec">
					<button
						id="login-btn"
						onClick={loggingIn}
						className={!isRegistering ? `selected`:``}
					>
						Log In
					</button>

					<button
						id="register-btn"
						onClick={registering}
						className={isRegistering ? `selected`:``}
					>
						Register
					</button>
				</div>

				<form onSubmit={handleSubmit}>
					<label>
						Email:
						<input
							type="text"
							value={email}
							onChange={handleEmailChange}
						/>
					</label>

					<label>
						Password:
						<input
							type="password"
							value={password}
							onChange={handlePasswordChange}
						/>
					</label>

					<input type="submit" value="Submit" />
				</form>
			</div>
		</div>
	);
};

LoginModal.propTypes = {
	isRegistering: PropTypes.bool
};

export default LoginModal;
