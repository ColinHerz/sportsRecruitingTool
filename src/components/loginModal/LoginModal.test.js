/* eslint-env jest */

import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "../../../enzyme.js";

import LoginModal from "./LoginModal.js";

const genProps = isRegistering => {
	return {
		logIn: jest.fn(),
		isRegistering: isRegistering,
		closeModal: jest.fn()
	};
};

describe(`rendering`, () => {
	beforeEach(() => {
		jest.resetModules();
	});

	it(`renders without first and last name when logging in`, () => {
		const props = genProps(false);

		const component = renderer.create(<LoginModal {... props} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it(`renders with first and last name when registering`, () => {
		const props = genProps(true);

		const component = renderer.create(<LoginModal {... props} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe(`functionality`, () => {

});
