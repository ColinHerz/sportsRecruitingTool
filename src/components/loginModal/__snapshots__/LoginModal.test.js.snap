// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders with first and last name when registering 1`] = `
<div
  id="gray-background"
>
  <div
    id="login-modal"
  >
    <h2>
      Sporta
    </h2>
    <button
      id="close"
      onClick={[Function]}
    >
      X
    </button>
    <div
      id="btn-sec"
    >
      <button
        className=""
        id="login-btn"
        onClick={[Function]}
      >
        Log In
      </button>
      <button
        className="selected"
        id="register-btn"
        onClick={[Function]}
      >
        Register
      </button>
    </div>
    <form
      onSubmit={[Function]}
    >
      <label>
        First Name:
        <input
          id="first-name"
          name="firstname"
          type="text"
        />
      </label>
      <label>
        Last Name:
        <input
          id="last-name"
          name="lastname"
          type="text"
        />
      </label>
      <label>
        Email:
        <input
          id="email"
          name="email"
          type="text"
        />
      </label>
      <label>
        Password:
        <input
          name="password"
          type="password"
        />
      </label>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`rendering renders without first and last name when logging in 1`] = `
<div
  id="gray-background"
>
  <div
    id="login-modal"
  >
    <h2>
      Sporta
    </h2>
    <button
      id="close"
      onClick={[Function]}
    >
      X
    </button>
    <div
      id="btn-sec"
    >
      <button
        className="selected"
        id="login-btn"
        onClick={[Function]}
      >
        Log In
      </button>
      <button
        className=""
        id="register-btn"
        onClick={[Function]}
      >
        Register
      </button>
    </div>
    <form
      onSubmit={[Function]}
    >
      <label>
        Email:
        <input
          id="email"
          name="email"
          type="text"
        />
      </label>
      <label>
        Password:
        <input
          name="password"
          type="password"
        />
      </label>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
